### PR TITLE
AccountStore: if database doesn't exist list no accounts instead of f…

### PR DESCRIPTION
**Pull request by danimoh**
…ailing

### CODE REVIEW

**Comment by sisou** [comment]
What is the reason for this change?

**Comment by sisou** [comment]
I know what it _does_, but _why_ does it need to be changed?

File: src/lib/AccountStore.js [hunk 1]
```diff
@@ -59,6 +60,7 @@ class AccountStore {
      */
     async list() {
         const db = await this.connect();
+        if (!db) return [];
```
Comment (svub) [inline src/lib/AccountStore.js]: I'd expect it to return `[]` in a promise though right away

File: src/lib/AccountStore.js [hunk 2]
```diff
@@ -96,6 +98,7 @@ class AccountStore {
      */
     async dangerousListPlain() {
         const db = await this.connect();
+        if (!db) return [];
```
Comment (svub) [inline src/lib/AccountStore.js]: same here. I always find it weird/ugly if a method returns a promise sometimes and an object other times. :/

Comment (svub) [reply src/lib/AccountStore.js]: So my suggestion is to move this `if` on line down and go `if (!db) resolve([]);`

Comment (sisou) [reply src/lib/AccountStore.js]: But it's an `async` function, so it returns a promise anyway, no?

Comment (danimoh) [reply src/lib/AccountStore.js]: yes, it returns a promise anyway and thus I'd say `return [];` is easier to read

**Comment by danimoh** [comment]
In my opinion the API is nicer if it doesn't throw if nothing unexpected happened.
E.g. if I call the method to list the accounts but there are none, then it should just return none and not throw.

**Pull request merged**